_posts/2024-11-15-WildFly-playing-with-generative-ai.adoc

```diff
@@ -0,0 +1,347 @@
+  * `in-memory-embedding-store`: provides integration with an in memory embedding store (for demo purpose only).
+  * `weaviate-embedding-store`: provides integration with https://weaviate.io/[Weaviate].
+* Support for content retriever as `ContentRetriever` for RAG:
+  * `default-embedding-content-retriever`: default content retriever using an `in-memory-embedding-store` and `in-memory-embedding-model-all-minilm-l6-v2` for embedding model.
+  * `web-search-engines`: provides support for Google and Tavily search engine.
+
+The layers will provide the required management resources, operations and modules.
+
+==== Provisioning the feature pack 
+
+For example:
+
+[source,xml]
+----
+<!-- The WildFly plugin deploys your war to a local JBoss AS container -->
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>${version.wildfly.maven.plugin}</version>
+    <configuration>
+        <feature-packs>
+            <feature-pack>
+                <location>org.wildfly:wildfly-galleon-pack:${version.wildfly.bom}</location>
+            </feature-pack>
+            <feature-pack>
+                <location>org.wildfly:wildfly-ai-galleon-pack:1.0.0-SNAPSHOT</location>
+            </feature-pack>
+        </feature-packs>
+        <layers
+            <!-- layers may be used to customize the server to provision-->
+            <layer>cloud-server</layer>
+            <layer>ollama-chat-model</layer>
+            <layer>default-embedding-content-retriever</layer>
+            <!-- providing the following layers -->
+            <!--
+                <layer>in-memory-embedding-model-all-minilm-l6-v2</layer>
+                <layer>in-memory-embedding-store</layer>
+            -->
+            <!-- Exisiting layers thart can be used -->
+            <!--
+                <layer>ollama-embedding-model</layer>
+                <layer>openai-chat-model</layer>
+                <layer>mistral-ai-chat-model</layer>
+                <layer>weaviate-embedding-store</layer>
+                <layer>web-search-engines</layer>
+            -->
+        </layers>
+        <name>ROOT.war</name>
+        <extraServerContentDirs>
+            <extraServerContentDir>extra-content</extraServerContentDir>
+        </extraServerContentDirs>
+                    <packagingScripts>
+            <packaging-script>
+                <scripts>
+                    <script>./src/scripts/configure_llm.cli</script>
+                </scripts>
+            </packaging-script>
+        </packagingScripts>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+In our example we are provisioning everything via layers.
+The script `configure_llm.cli`provides sample commands to set up manually the subsystem.
+
+
+[source,shell]
+----
+
```

Comment:
Not necessary, but a thought...

Unless the commands below assume the presence of detail resources generated by the layers used above, perhaps start with the extension add and subsystem add commands, with the comment explaining they aren't needed if the extension/subsystem are already in the config.

Reply:
Without the layers the modules wouldn't be provisionned